### ISSUE TITLE
Automatically run OptiPNG on PNGs uploaded to any Flutter repo.

### ISSUE THREAD

**Issue by matthew-carroll** [flutter/flutter#20548]
Automatically run OptiPNG on PNGs uploaded to any Flutter repo.

Every image a developer uploads to any Flutter repo (framework, website, examples, etc) needs to go through OptiPNG. Currently this is left to individual developers before putting up a PR.  We should integrate this as either a pre or post commit hook on PRs so that developers don't have to worry about this, and reviewers don't have to police it.

CC @xster

**Comment by Sfshaza** [comment]
I thought we *were* doing that, but maybe only on the *.dartlang sites. @chalin?

**Comment by chalin** [comment]
Yes, we have scripts for the *.dartlang sites that sweep over all images, but had decided to not  integrate them into the build checks / or set them up via hooks. IMHO, it is probably enough to setup a cron job to check this offline, outside of the main site-update workflow.

I've created https://github.com/flutter/website/issues/1162 to track this for flutter/website.

**Comment by zoeyfan** [comment]
@matthew-carroll , I am triaging our bug database, and noticed that this bug was opened in 2018, but had no traction afterwards. Just curious if we are still planning to do anything here. If not, shall we just go ahead and close it?

**Comment by xster** [comment]
I think this is still relevant and would save another mechanical review step for contributors.

**Comment by kf6gpe** [comment]
As this is an improvement for the web site, moving to the web site repo.
